packages/cli/src/lib/webpack/webpack-base-config.js

```diff
@@ -357,12 +357,10 @@ module.exports = function createBaseConfig(env) {
 			}),
 			...(tsconfig
 				? [
-						new ForkTsCheckerWebpackPlugin({
-							checkSyntacticErrors: true,
-							async: !isProd,
-							tsconfig: tsconfig,
-							silent: !isWatch,
-						}),
+					new ForkTsCheckerWebpackPlugin({
+						typescript: {configFile: tsconfig, diagnosticOptions: {semantic: true, syntactic: true} },
```

Comment:
`semantic: true` is not something we want to set. 

Additionally, this should be formatted. We use precommit hooks via Husky to do this, generally.

Reply:
> `semantic: true` is not something we want to set.
> 
I just made it explicit, but `semantic: true` is already the default also in the forkstsc 7.x branch used in pract-cli 4.0.
Just wanted to make sure you are aware of this, and you're just meaning that it's not needed to make it explicit, and we can leave it to the default value.


> Additionally, this should be formatted. We use precommit hooks via Husky to do this, generally.
I'm not familiar with Husky, but I will look into it 😄

Reply:
> I just made it explicit, but semantic: true is already the default also in the forkstsc 7.x branch used in pract-cli 4.0.

Well, no, you've changed the option there. While I decided to change that in the new major, that should not be backported (IMO) to 3.x

>  I'm not familiar with Husky, but I will look into it 😄

It's just a precommit hook. I imagine you didn't clone the repo locally to make these changes? You can also format manually using our prettier config.

Reply:
> Well, no, you've changed the option there. While I decided to change that in the new major, that should not be backported (IMO) to 3.x
> 

uhm, I get your point, but we should already be ok then, please see here https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/blob/v4.1.6/README.md (this is the version you were using before). From what I can understand, the old version is already enabling semantic checks by default? 
I quote from their README (```checkSyntacticErrors boolean: This option is useful if you're using ts-loader in happyPackMode with [HappyPack](https://github.com/amireh/happypack) or [thread-loader](https://github.com/webpack-contrib/thread-loader) to parallelise your builds. If true it will ensure that the plugin checks for both syntactic errors (eg const array = [{} {}];) and semantic errors (eg const x: number = '1';). By default the plugin only checks for semantic errors. This is because when ts-loader is used in transpileOnly mode, ts-loader will still report syntactic errors. When used in happyPackMode it does not. Default: false.``` It's a bit tricky to districate between changes in other plugins, defaults, etc.etc. What do you think? 

> > I'm not familiar with Husky, but I will look into it 😄
> 
> It's just a precommit hook. I imagine you didn't clone the repo locally to make these changes? You can also format manually using our prettier config.
Well, I cloned it, but didn't install anything. Now it's all clear (should be 😄 )

Reply:
>  From what I can understand, the old version is already enabling semantic checks by default?

You're right, my bad!